### PR TITLE
Focused-Launch: Update <Title/>, <SubTitle/> & <p/> to improve a11y

### LINKED ISSUE #47497: Focused Launch: replace custom code with <Title> and <Subtitle> components from @automattic/onboarding
Once  #47418 and #47496 are implemented, code in the Focused Launch screens should be refactored to re-use `<Title>` and `<Subtitle>` components as much as possible.

Particular attention should be applied to make sure that:
- the document outline is correct (i.e. since the modal already comes with a `<h1>` tag, all titles/subtitles in the Focused Launch views should start from `<h2>`)
- the classnames added from the Focused Launch point of view are applied and merged correctly

### PR DESCRIPTION
**Pull request by StefanNieuwenhuis**
The focused launch modal has to be semantically correct to enable all users to enjoy its features, hence we added tagName props to the aforementioned elements to improve semantics. We also converted `<p/>` to `<SubTitle/>` where necessary to improve consistency.

#### Changes proposed in this Pull Request

* Add `tagName` props to `<Title/>`, `<SubTitle/>` & `<p/>`;
* Convert `<p/>` to `<SubTitle/>`.

#### Testing instructions

* Check out this branch on your machine and run `yarn && yarn start`;
* Navigate to the focused launch flow;
* Click on "View all plans";
* use `devTools` to inspect the "Select a plan" heading and confirm that's a `<h2/>`.
* use `devTools` to inspect the "There's no risk, you can cancel for a full refund within 30 days." subtitle and confirm that's a `<h3/>`.

Fixes #47497

### CODE REVIEW

File: packages/launch/src/focused-launch/summary/index.tsx [hunk 1]
```diff
@@ -130,8 +130,10 @@ const DomainStep: React.FunctionComponent< DomainStepProps > = ( {
 							>
 								{ info }
 							</Tooltip>
-							{ /* @TODO: should use <Subtitle/> once https://github.com/Automattic/wp-calypso/issues/47418 is solved */ }
-							<p className="focused-launch-summary__mobile-commentary focused-launch-summary__mobile-only">
+							<SubTitle
+								tagName="p"
+								className="focused-launch-summary__mobile-commentary focused-launch-summary__mobile-only"
+							>
```
Comment (ciampo) [inline packages/launch/src/focused-launch/summary/index.tsx:136]: This `<p>` tag actually didn't need to be converted to `<Subtitle />` — sorry, my bad!
Please revert the change (but still delete the TODO comment)

File: packages/launch/src/focused-launch/summary/index.tsx [hunk 2]
```diff
@@ -518,12 +520,12 @@ const Summary: React.FunctionComponent = () => {
...
 		<div className="focused-launch-summary__container">
 			<div className="focused-launch-summary__section">
-				<Title>
+				<Title tagName="h2">
 					{ hasPaidDomain && hasPaidPlan
 						? __( "You're ready to launch", __i18n_text_domain__ )
 						: __( "You're almost there", __i18n_text_domain__ ) }
 				</Title>
-				<p className="focused-launch-summary__caption">
+				<SubTitle tagName="p" className="focused-launch-summary__caption">
```
Comment (ciampo) [inline packages/launch/src/focused-launch/summary/index.tsx:528]: Could also remove the `font-size` and `color` CSS rules from the `.focused-launch-summary__caption` ? We don't need them anymore, now that we're using `<Subtitle />`

**Review by ciampo — COMMENTED** [review]
Thank you for working on these changes! I just left a couple of comments, but overall it looks good!

File: packages/launch/src/focused-launch/summary/index.tsx (hunk 1 above)
Comment (StefanNieuwenhuis) [reply packages/launch/src/focused-launch/summary/index.tsx]: I just reverted the changes and left the comment out.

File: packages/launch/src/focused-launch/summary/index.tsx (hunk 2 above)
Comment (StefanNieuwenhuis) [reply packages/launch/src/focused-launch/summary/index.tsx:525]: Makes total sense and I removed them!

**Comment by StefanNieuwenhuis** [comment]
> Thank you for working on these changes! I just left a couple of comments, but overall it looks good!

Thanks for reviewing!

**Review by ciampo — APPROVED** [review]
Thank you for reacting quickly to the feedback!

LGTM
